### PR TITLE
Whatsapp Automated Message

### PR DESCRIPTION
**Pull request by sayantani11**
Here's the code.....I hope it will be working fine
for issue 
Fixes: #588

### CODE REVIEW

File: Whatsapp_Automated_Messages/__main__.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    pt.moveRe(12,15)
+    pt.click()
+    whatsapp_message = pyperclip.paste()
+    pt.click()
+    print("Message received: " + whatsapp_message)
+    return whatsapp_message
+
+def post_response(message):               # to post our desire response
+    global x, y
+    postion= pt.locateOnScreen("Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
```
Comment (antrikshmisri) [inline Whatsapp_Automated_Messages/__main__.py:32]:
```suggestion
    poistion= pt.locateOnScreen("Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
```

File: Whatsapp_Automated_Messages/__main__.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        return "It's Okay"
+
+    elif "Thanks" in str(message).lower():
+        return "Mention Not"
+    
+    else:
+        if random_no==0:
+            return "Talk to you later"
+
+        elif randm_no==1:
```
Comment (antrikshmisri) [inline Whatsapp_Automated_Messages/__main__.py:66]:
```suggestion
        elif random_no==1:
```

File: Whatsapp_Automated_Messages/__main__.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                pt.moveRel(-100,0)
+                pt.click()
+                time.sleep(2)
+
+        except(Exception):
+            print("No new messages")
+
+        if pt.pixelMatchesColor(int(x+50),int(y-35),(255,255,255), tolerance=10):
+            print("is white")
+            processed_message= process_response(get_message())
```
Comment (antrikshmisri) [inline Whatsapp_Automated_Messages/__main__.py:90]:
```suggestion
            processed_message= process_responses(get_message())
```

**Review by antrikshmisri — CHANGES_REQUESTED** [review]
@sayantani11 Please run/test your code before pushing. Fix the typos as mentioned above

**Comment by sayantani11** [comment]
I guess it's fine now

File: Whatsapp_Automated_Messages/ReadME.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+AIM
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/ReadME.md:1]: Format this readme
- All headings need to have levels. Level 2 (##)

File: Whatsapp_Automated_Messages/__main__.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
+import pyautogui as pt 
+import time
+import pyperclip
+import random 
+
+time.sleep(4)
+
+position1 = pt.locateOnScreen("Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/__main__.py:8]:
```suggestion
position1 = pt.locateOnScreen("./Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
```

File: Whatsapp_Automated_Messages/__main__.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+x= position1[0]
+y= position1[1]
+
+
+def get_message():             # it will help us to fetch the messages
+    global x, y
+
+    position = pt.locateOnScreen("Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/__main__.py:16]: you have defined position1 before, isn't that same?

File: Whatsapp_Automated_Messages/__main__.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    pt.moveRe(12,15)
+    pt.click()
+    whatsapp_message = pyperclip.paste()
+    pt.click()
+    print("Message received: " + whatsapp_message)
+    return whatsapp_message
+
+def post_response(message):               # to post our desire response
+    global x, y
+    position= pt.locateOnScreen("Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/__main__.py:32]: this also

File: Whatsapp_Automated_Messages/__main__.py [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+        else:
+            return "A bit busy!"
+
+def new_message_check():       # to keep checking if new message is received
+    pt.moveTo(x+50, y-35)
+
+    while True:
+        try:
+            position= pt.locateOnScreen("Whatsapp_Automated_Messsages/green_mark.png", confidence=.6)
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/__main__.py:77]:
```suggestion
            position= pt.locateOnScreen("./Whatsapp_Automated_Messsages/green_mark.png", confidence=.6)
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Whatsapp_Automated_Messages/__main__.py (hunk 6 above)
Comment (sayantani11) [reply Whatsapp_Automated_Messages/__main__.py:16]: No actually not

File: Whatsapp_Automated_Messages/ReadME.md (hunk 4 above)
Comment (sayantani11) [reply Whatsapp_Automated_Messages/ReadME.md:1]: Where am I support to add the level??

Comment (kaustubhgupta) [reply Whatsapp_Automated_Messages/ReadME.md:1]: ```## Script name```

File: Whatsapp_Automated_Messages/__main__.py (hunk 6 above)
Comment (kaustubhgupta) [reply Whatsapp_Automated_Messages/__main__.py:16]:
```python
position1 = pt.locateOnScreen("./Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
position = pt.locateOnScreen("./Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
```
What's the difference?

Comment (sayantani11) [reply Whatsapp_Automated_Messages/__main__.py:16]: I don't know how to put it into words
When I was writing the function I felt its necessary to introduce a new variable to improve readability, improve how I interact with my own code, and more importantly the two variables here have values that do very different functions and it was needed to introduce two variables and not one.

Comment (kaustubhgupta) [reply Whatsapp_Automated_Messages/__main__.py]: You can make position variable global which can be used anywhere in the function. Second, duplicate variables doesn't make code readable

File: Whatsapp_Automated_Messages/ReadME.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
+## WHATSAPP AUTOMATED MESSAGE --- LEVEL 2
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/ReadME.md:1]:
```suggestion
## WHATSAPP AUTOMATED MESSAGE 
```

File: Whatsapp_Automated_Messages/ReadME.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+# AIM
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/ReadME.md:3]: This and other headings after this needs to be level 2 (##)

File: Whatsapp_Automated_Messages/__main__.py [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+position1 = pt.locateOnScreen("Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
+x= position1[0]
+y= position1[1]
+
+
+def get_message():             # it will help us to fetch the messages
+    global x, y
+
+    position = pt.locateOnScreen("./Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
+    x=position[0]
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/__main__.py:17]: this x, y is also repeated throughout the code. Until these values are not changing, there is no need to define them again and again

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by sayantani11** [comment]
The code faces issue once I make use of global variables I guess.

**Comment by kaustubhgupta** [comment]
> The code faces issue once I make use of global variables I guess.

It's because of position1 https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/652/checks?check_run_id=2167226505

**Comment by sayantani11** [comment]
> > The code faces issue once I make use of global variables I guess.
> 
> It's because of position1 https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/652/checks?check_run_id=2167226505

I am not getting it actually.

**Comment by kaustubhgupta** [comment]
> > > The code faces issue once I make use of global variables I guess.
> > 
> > 
> > It's because of position1 https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/652/checks?check_run_id=2167226505
> 
> I am not getting it actually.

Click here to get the error report and see what errors you are making
image

**Comment by sayantani11** [comment]
> > > > The code faces issue once I make use of global variables I guess.
> > > 
> > > 
> > > It's because of position1 https://github.com/avinashkranjan/Amazing-Python-Scripts/pull/652/checks?check_run_id=2167226505
> > 
> > 
> > I am not getting it actually.
> 
> Click here to get the error report and see what errors you are making
> image

I did.....But it shows the log
How to see the problem in the code?

**Comment by kaustubhgupta** [comment]
Scroll up and here you will have errors for every file(if exist)
image

**Comment by sayantani11** [comment]
I guess it should be fine now.....When I removed the position1 the error showed
When I make changes I guess there is some problem
which I am unable to figure out

**Comment by kaustubhgupta** [comment]
> I guess it should be fine now.....When I removed the position1 the error showed
> When I make changes I guess there is some problem
> which I am unable to figure out

Umm, okay, I will re-review your code in the morning and will let you know if there is any change else I will approve

File: Whatsapp_Automated_Messages/ReadME.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
+## WHATSAPP AUTOMATED MESSAGE 
+## LEVEL 2
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/ReadME.md:2]: Remove this line

File: Whatsapp_Automated_Messages/ReadME.md [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+# HOW TO USE:    
+This an automated message bot as of which you need to add this to your app and the rest will be done by it.
+   
+# MODULES USED:   
+Pyautogui
+Time
+Pyperclip 
+Random
+OpenCV
```
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/ReadME.md:16]: no opencv has been used, remove this

File: Whatsapp_Automated_Messages/__main__.py (hunk 5 above)
Comment (kaustubhgupta) [inline Whatsapp_Automated_Messages/__main__.py:8]:
```suggestion
position1 = pt.locateOnScreen("./Whatsapp_Automated_Messsages/smilie_paperclip.png", confidence=.6)
```

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
Here is the final review after testing the script locally: (Read all of them)
- I understood your point of defining the x,y, and position in every function, that's okay no issues there
- You have not used selenium automation that prompts to open the WhatsApp web page and then authenticate via QR code. It makes no sense to open this page before running the script and then wait for the bot to reply. Even if I open the WhatsApp Web before script, your code is not working:
image

**Comment by kaustubhgupta** [comment]
@sayantani11 you have not updated the code? After this, I will be forced to close this PR

**Comment by sayantani11** [comment]
I guess you can close the issue, I am unable to solve the problem.

**Pull request closed without merging**